templates/single-product/product-attributes.php

```diff
@@ -41,7 +41,25 @@
 		<tr>
 			<th><?php echo wc_attribute_label( $attribute->get_name() ); ?></th>
 			<td><?php
-				$values = $attribute->is_taxonomy() ? wc_get_product_terms( $product->get_id(), $attribute->get_name(), array( 'fields' => 'names' ) ) : $attribute->get_options();
+				$values = array();
+
+				if ( $attribute->is_taxonomy() ) :
+
+					$attribute_taxonomy = $attribute->get_taxonomy_object();
+					$attribute_values = wc_get_product_terms( $product->get_id(), $attribute->get_name(), array( 'fields' => 'all' ) );
+
+					foreach ( $attribute_values as $attribute_value ) :
+						if ( $attribute_taxonomy->attribute_public ) :
+							$values[] = '<a href="' . esc_url( get_term_link( $attribute_value->term_id, $attribute->get_name() ) ) . '" rel="tag">' . $attribute_value->name . '</a>';
```

Comment:
Definitely. `esc_html` is the way to go, since attribute values aren't supposed to have links or other tags in their name.